cmd/lncli/commands.go

```diff
@@ -688,53 +695,161 @@ func closeChannel(ctx *cli.Context) error {
-			})
+			closingTxid = txid.String()
 
 			if !ctx.Bool("block") {
-				return nil
+				return closingTxid, nil
 			}
-
 		case *lnrpc.CloseStatusUpdate_ChanClose:
 			closingHash := update.ChanClose.ClosingTxid
 			txid, err := chainhash.NewHash(closingHash)
 			if err != nil {
-				return err
+				return "", err
 			}
 
-			printJSON(struct {
-				ClosingTXID string `json:"closing_txid"`
-			}{
-				ClosingTXID: txid.String(),
-			})
+			closingTxid = txid.String()
 		}
 	}
 }
 
+func closeAllChannels(ctx *cli.Context) error {
+	ctxb := context.Background()
+	client, cleanUp := getClient(ctx)
+	defer cleanUp()
+
+	listReq := &lnrpc.ListChannelsRequest{}
+	openChannels, err := client.ListChannels(ctxb, listReq)
+	if err != nil {
+		return fmt.Errorf("unable to fetch open channels: %v", err)
+	}
+
+	// result defines the result of closing a channel. The closing txid will
+	// be populated if the channel was closed successfully. Otherwise, the
+	// error that prevented us from closing a channel will be populated.
+	type result struct {
+		RemotePubKey string `json:"remote_pub_key"`
+		ChannelPoint string `json:"channel_point"`
+		ClosingTxid  string `json:"closing_txid"`
+		FailErr      string `json:"error"`
+	}
+
+	results := make([]result, len(openChannels.Channels))
+	for i, channel := range openChannels.Channels {
+		res := result{}
+		res.RemotePubKey = channel.RemotePubkey
+		res.ChannelPoint = channel.ChannelPoint
+
+		// If the channel is inactive, we will attempt to force close
+		// the channel and prompt the user for confirmation.
+		forceClose := !channel.GetActive()
+		if forceClose {
+			fmt.Printf("Force close channel with node %s? "+
+				"The closing transaction will need %d "+
+				"confirmations before the funds can be "+
+				"accessed. (y/N): ", res.RemotePubKey,
+				channel.CsvDelay)
+
+			reader := bufio.NewReader(os.Stdin)
+			answer, err := reader.ReadString('\n')
+			if err != nil {
+				res.FailErr = fmt.Sprintf("unable to read "+
+					"input string: %v", err)
+				results[i] = res
+				continue
+			}
+			answer = strings.ToLower(strings.TrimSpace(answer))
+
+			// Deny any input other than 'y' or 'yes'.
+			if !(answer == "y" || answer == "yes") {
+				continue
+			}
+		}
+
+		// Parse the channel point in order to create the close channel
+		// request.
+		s := strings.Split(res.ChannelPoint, ":")
+		if len(s) != 2 {
+			res.FailErr = "expected channel point with format " +
+				"txid:index"
+			results[i] = res
+			continue
+		}
+		index, err := strconv.ParseUint(s[1], 10, 32)
+		if err != nil {
+			res.FailErr = fmt.Sprintf("unable to parse channel "+
+				"point output index: %v", err)
+			results[i] = res
+			continue
+		}
+
+		req := &lnrpc.CloseChannelRequest{
+			ChannelPoint: &lnrpc.ChannelPoint{
+				FundingTxid: &lnrpc.ChannelPoint_FundingTxidStr{
+					FundingTxidStr: s[0],
+				},
+				OutputIndex: uint32(index),
+			},
+			Force: forceClose,
+		}
+
+		closingTxid, err := executeChannelClose(ctx, req)
```

Comment:
I haven't tested this change, but won't this mean that you will have to wait for one `executeChannelClose` to finish before going to the next one? I suspect this can get slow when you have many channels, some of which must be force closed. You also won't see any of the results before all channels finishes closing. 

Possible to ask for confirmation up front, then closing all channels in parallel?